modules/core/include/opencv2/core/mat.hpp

```diff
@@ -1084,6 +1084,18 @@ class CV_EXPORTS Mat
     */
     void copyTo( OutputArray m, InputArray mask ) const;
 
+#if CV_FP16
```

Comment:
I just moved the function out of cv::Mat.

I figured out that in some cases on ARM 32bit platform, segmentation fault error occurs due to the memory alignment.  So for this, I have no choice but to implement the SW version.  I'll make time and commit a SW version later.